repair_shards.c

```diff
@@ -137,10 +137,7 @@ master_copy_shard_placement(PG_FUNCTION_ARGS)
 	}
 
 	/* the placement is repaired, so return to finalized state */
-	DeleteShardPlacementRow(targetPlacement->id);
```

Comment:
Yes. CitusDB has fields `pg_shard` doesn't know about (shard length or whatnot). If we do `DELETE`/`INSERT`, `pg_shard` must know about them. If we just do an `UPDATE`, we can update the fields we care about without having to keep track of and re-insert the ones we don't.